core/trino-main/src/test/java/io/trino/sql/query/TestCheckConstraint.java

```diff
@@ -187,34 +187,34 @@ public void testInsert()
     public void testMergeInsert()
     {
         // Within allowed check constraint
-        assertThatThrownBy(() -> assertions.query("""
+        assertThat(assertions.query("""
```

Comment:
Please separate the Delta Lake related part in a subsequent commit.

This commit is about the engine (the fact that it gets constraint to work on Delta Lake is only a side effect).